### PR TITLE
2654936 exceptions

### PR DESCRIPTION
**Pull request by Cryt1c**
See https://www.drupal.org/node/2654936

I have merged 8.x-3.x into the stalled pull request (https://github.com/fago/rules/pull/369) and fixed some more global LogicExceptions.
There is a problems when I change the InvalidArgumentException in DataFetcherTest.php and DataFetcher.php to the rules namespace. A global \InvalidArgumentException is thrown further down and therefor also expected in the DataFetcherTest.php. Should this \InvalidArgumentException be encapsuled by a rules InvalidArgumentException?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/.idea
```
Comment (klausi) [inline .gitignore]: I don't think we should add this file with this pull request. You can have that file locally and ignore itself even.

File: d7-tests/rules.test [hunk 2]
```diff
@@ -191,8 +191,8 @@ class RulesTestCase extends DrupalWebTestCase {
       $action->execute();
       $this->fail("Execution hasn't created an exception.");
     }
-    catch (RulesEvaluationException $e) {
-      $this->pass("RulesEvaluationException was thrown: ". $e);
+    catch (EvaluationException $e) {
+      $this->pass("EvaluationException was thrown: ". $e);
```
Comment (klausi) [inline d7-tests/rules.test]: This is the old Drupal 7 file and should NOT be modified. It is only there for reference purposes and to compare.

File: src/TypedData/DataFetcher.php [hunk 3]
```diff
@@ -98,6 +99,7 @@ public function fetchDataBySubPaths(TypedDataInterface $typed_data, array $sub_p
       $current_selector = implode('.', $current_selector);
       throw new \InvalidArgumentException("Unable to apply data selector '$selector' at '$current_selector': " . $e->getMessage());
```
Comment (klausi) [inline src/TypedData/DataFetcher.php]: here is another global \InvalidArgumentException that you forgot? Make sure to grep the code base for all remaining global exceptions

File: src/TypedData/DataFetcher.php [hunk 4]
```diff
@@ -98,6 +99,7 @@ public function fetchDataBySubPaths(TypedDataInterface $typed_data, array $sub_p
...
     }
+
```
Comment (klausi) [inline src/TypedData/DataFetcher.php]: Unrelated change that can be removed?

**Comment by fago** [comment]
see #457
